dissect/cstruct/types/pointer.py

```diff
@@ -16,7 +16,7 @@ class Pointer(int, BaseType):
     _context: dict[str, Any]
     _value: BaseType
 
-    def __new__(cls, value: int, stream: BinaryIO, context: dict[str, Any] = None) -> Pointer:
+    def __new__(cls, value: int, stream: BinaryIO | None, context: dict[str, Any] = None) -> Pointer:
```

Comment:
(curious)

If `context` can be `None`, should the type not reflect this, i.e. `dict[str, Any] | None`?
This would propagate to the `_context` field, and possibly further.

Perhaps an empty dictionary is safer than `None`, although possibly somewhat wasteful?